Tensile/KernelWriter.py

```diff
@@ -1273,8 +1293,10 @@ def noLoadLoop( self, kernel, tensorParametersA, tensorParametersB, isOptNLL, is
             self.globalReadIncrements.addCode(Code.Module("globalReadIncrementA"))
             self.globalReadIncrements.addCode(Code.Module("globalReadIncrementB"))
         if not isLastLoop:
-          self.localWriteACode = self.localWriteDo(kernel, tensorParametersA, (uDu+1)%kernel["DepthULdsDivisor"])  # local write in loopcnt N targets data for loopcnt N+1
-          self.localWriteBCode = self.localWriteDo(kernel, tensorParametersB, (uDu+1)%kernel["DepthULdsDivisor"])
+          self.localWriteACode, tmpCheckedOutVgprA = self.localWriteDo(kernel, tensorParametersA, (uDu+1)%kernel["DepthULdsDivisor"])  # local write in loopcnt N targets data for loopcnt N+1
+          self.localWriteBCode, tmpCheckedOutVgprB = self.localWriteDo(kernel, tensorParametersB, (uDu+1)%kernel["DepthULdsDivisor"])
+          self.tmpCheckedOutLWVgprs.append(tmpCheckedOutVgprA)
+          self.tmpCheckedOutLWVgprs.append(tmpCheckedOutVgprB)
```

Comment:
There have been some thoughts about using an RAII approach to temp registers within the allocation pool. This has been loosely done with sgprs  with KernelWriterAssembly.TmpSgpr. (Though would be better to override __enter__ and __exit__ for explicit cleanup).

Idea is that on instantiation registers are checked out and on deletion registers are checked in. This way the registers at least are cleaned up once they go out of scope and we don't manually have to check them in.

This might be enough scope for a separate ticket, but maybe we can work on this in a different ticket. I just wanted to get the idea out there.

Reply:
Yes, I agree on the tempReg is not really  convenient to use, and tmpGpr could be overwriten when nested scope..Maybe we can have another task to refine the regPool